frontend/src/metabase/common/components/DataPicker/utils.ts

```diff
@@ -156,22 +163,26 @@ export const createShouldShowItem = (
   models: CollectionItemModel[],
   databaseId?: DatabaseId,
 ) => {
-  return (item: QuestionPickerItem) => {
+  return (item: QuestionPickerItem & { database_id?: DatabaseId }) => {
     if (item.model === "collection") {
       const below = item.below ?? [];
       const here = item.here ?? [];
       const contents = [...below, ...here];
-      const hasCards = models.some(model => contents.includes(model));
+      const hasCards = models.some(model =>
+        contents.includes(model as CollectionItemModel),
+      );
 
       if (item.id !== "root" && !item.is_personal && !hasCards) {
         return false;
+      } else {
+        return true;
```

Comment:
To make this even simpler, maybe you could do:

```ts
return (item.id === "root" || item.is_personal || hasCards)
```

Reply:
Ha. Good find!

Reply:
I've actually updated this again. This logic wasn't quite right, and led to the root item list being hidden haha. But I think it's sorted now